android/java/org/chromium/chrome/browser/crypto_wallet/activities/NftDetailActivity.java

```diff
@@ -113,13 +123,22 @@ protected void triggerLayoutInflation() {
         mNetworkNameView = findViewById(R.id.blockchain_content);
 
         mTokenStandardView = findViewById(R.id.token_standard_content);
-        mTokenStandardView.setText(R.string.brave_wallet_nft_erc_721);
 
-        mTokenIdView = findViewById(R.id.token_id_content);
+        if (mIsErc721) {
+            mTokenStandardView.setText(R.string.brave_wallet_nft_erc_721);
+        } else {
+            mTokenStandardView.setText(R.string.brave_wallet_nft_sol_spl);
+        }
+
+        mTokenAddressLabelView = findViewById(R.id.token_address_label);
+        if (!mIsErc721) {
+            mTokenAddressLabelView.setText(R.string.brave_wallet_nft_mint_address);
```

Comment:
nit: Not sure if all non-ERC21 NFTs should have a mint address, might wanna discuss with @jamesmudgett if required.

Reply:
Excellent observation. 
I'll wait for James opinion and also check the approach taken by the other platform.